pkg/reconcilers/deployment_mutator.go

```diff
@@ -88,17 +93,63 @@ func DeploymentImageMutator(desired, existing *k8sapps.Deployment) bool {
 	return update
 }
 
-func DeploymentVolumesMutator(desired, existing *k8sapps.Deployment) bool {
+func DeploymentImagePullPolicyMutator(desired, existing *k8sapps.Deployment) bool {
 	update := false
 
-	if !reflect.DeepEqual(existing.Spec.Template.Spec.Volumes, desired.Spec.Template.Spec.Volumes) {
-		existing.Spec.Template.Spec.Volumes = desired.Spec.Template.Spec.Volumes
+	if existing.Spec.Template.Spec.Containers[0].ImagePullPolicy != desired.Spec.Template.Spec.Containers[0].ImagePullPolicy {
```

Comment:
Should we nil check containers here to ensure we don't panic ?

Reply:
well, yes and no. 

Obviously, yes, the code should check. Other mutators do not, but that's not a really good reason to not do it. On the other hand, the `DeploymentContainerListMutator` comes first and makes sure that there number of containers is the same, and the desired deployment has at least one container.. so,... adding those checks but not really needed IMO

Reply:
done